---
title: "Initial check of RNA-seq data"
date: 2016-05-18
output:
  html_document:
    toc: true
    toc_float: true
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options}
library("knitr")
opts_chunk$set(cache = FALSE)
```

Initial quick analysis of RNA-seq data.

## Setup

```{r packages, message=FALSE}
library("dplyr")
library("tidyr")
library("ggplot2")
library("cowplot")
theme_set(theme_bw(base_size = 12))
```


```{r}
info <- read.delim("../data/experiment-info.txt", stringsAsFactors = FALSE)
tpm_raw <- read.delim("../data/tpm.txt", stringsAsFactors = FALSE)
```

Prepare data files.

```{r}
gene_info <- tpm_raw %>% select(ensembl_gene_id, chromosome_name,
                                external_gene_name, gene_biotype)
tpm <- tpm_raw %>% select(starts_with("X"))
anno <- data.frame(id = colnames(tpm))
anno <- anno %>% mutate(id = gsub("\\.", "-", id),
                        id = substr(id, 2, nchar(id)))
anno <- separate(anno, id, into = c("individual", "susceptibility", "treatment",
                                    "flow_cell", "lane"), sep = "-", remove = FALSE)
anno <- anno %>% mutate(individual = paste0("i", individual))
head(anno)
```

## Remove lowly expressed genes

Median log2 tpm.

```{r}
tpm_log2_median <- apply(log2(tpm + 1), 1, median)
hist(tpm_log2_median)
cutoff <- 2
abline(v = cutoff, col = "red")
```

Keeping the `r sum(tpm_log2_median > cutoff)` genes with a median log2 tpm greater than `r cutoff`.

```{r}
tpm_clean <- tpm %>% filter(tpm_log2_median > cutoff)
```

## PCA

```{r}
pca <- prcomp(t(tpm_clean), scale. = TRUE)
variances <- pca$sdev^2
explained <- variances / sum(variances)
plot(pca)
```

```{r}
pca_data <- cbind(anno, pca$x[, 1:5])
```

PC1 versus PC2.

```{r pc1-pc2}
pc1pc2 <- ggplot(pca_data, aes(x = PC1, y = PC2)) +
  geom_text(aes(label = individual)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)))
pc1pc2
```

```{r pc1-pc2-treatment}
pc1pc2 %+% aes(color = treatment) + labs(title = "Treatment")
```

```{r pc1-pc2-susceptibility}
pc1pc2 %+% aes(color = susceptibility) + labs(title = "Susceptibility")
```

```{r pc1-pc2-lane}
pc1pc2 %+% aes(color = lane) + labs(title = "Lane")
```

PC3 versus PC4.

```{r pc3-pc4}
pc3pc4 <- pc1pc2 %+% aes(x = PC3, y = PC4) +
    labs(x = sprintf("PC%d (%.2f%%)", 3, round(explained[3] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 4, round(explained[4] * 100, 2)))
pc3pc4
```

```{r pc3-pc4-treatment}
pc3pc4 %+% aes(color = treatment) + labs(title = "Treatment")
```

```{r pc3-pc4-susceptibility}
pc3pc4 %+% aes(color = susceptibility) + labs(title = "Susceptibility")
```

```{r pc3-pc4-lane}
pc3pc4 %+% aes(color = lane) + labs(title = "Lane")
```

## Hierarchical clustering

```{r dendrogram}
d <- dist(t(tpm_clean))
h <- hclust(d)
plot(h, labels = anno$id)
```

## Session information

```{r info}
sessionInfo()
```
