---
title: "Directly compare to lbb2012"
date: 2016-07-07
output:
  html_document:
    toc: true
    toc_float: true
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options}
library("knitr")
opts_chunk$set(cache = FALSE, fig.pos = "center", fig.width = 7, fig.height = 7,
               dev = c("png", "postscript"))
```

## Setup

```{r packages, message=FALSE}
library("limma")
library("edgeR")
library("ashr")
library("caret")
library("glmnet")
library("ggplot2")
library("ggdendro")
library("cowplot")
```

Input data.

```{r input-counts}
counts <- read.delim("../data/counts-filtered.txt", check.names = FALSE, row.names = 1)
info <- read.delim("../data/experiment-info-filtered.txt",
                   stringsAsFactors = FALSE, row.names = 1)
info$study <- "current"
stopifnot(colnames(counts) == rownames(info))
```

Calculate counts per million.

```{r calc-cpm}
counts_cpm <- cpm(counts, log = TRUE)
```

Input Barreiro et al., 2012 data.

```{r lbb2012}
load("../data/Exp_final_Batch_corrected.Rdata")
lbb2012 <- Exp_final_Batch_corrected[, -1:-2]
rownames(lbb2012) <- Exp_final_Batch_corrected$Ensembl_ID
id <- sub("Mtb", "", colnames(lbb2012))
id <- sub("_", "", id)
id <- sub("_", "", id)
id <- sub("\\.neg", "", id)
id <- sub("\\.pos", "", id)
id <- sub("\\.TB", "", id)
id <- sub("\\.1", "", id)
id <- sub("\\.2", "", id)
info_lbb <- data.frame(individual = id, status = "unknown", study= "lbb2012",
                       treatment = ifelse(grepl("neg", colnames(lbb2012)),
                                              "none", "infected"))
colnames(lbb2012) <- paste(info_lbb$individual,
                           info_lbb$status,
                           info_lbb$treatment, sep = "-")
rownames(info_lbb) <- colnames(lbb2012)
stopifnot(ncol(lbb2012) == 65 * 2, table(id) == 2)
head(info_lbb)
```

Subset to only include genes common to the two studies.

```{r common-genes}
common_genes <- intersect(rownames(counts_cpm), rownames(lbb2012))
length(common_genes)
counts_cpm_sub <- counts_cpm[rownames(counts_cpm) %in% common_genes, ]
counts_cpm_sub <- counts_cpm_sub[order(rownames(counts_cpm_sub)), ]
lbb2012_sub <- lbb2012[rownames(lbb2012) %in% common_genes, ]
lbb2012_sub <- lbb2012_sub[order(rownames(lbb2012_sub)), ]
stopifnot(nrow(counts_cpm_sub) == length(common_genes),
          rownames(counts_cpm_sub) == rownames(lbb2012_sub))
```

Compare the distributions.

```{r distributions}
rnaseq_medians <- apply(counts_cpm_sub, 1, median)
summary(rnaseq_medians)
plot(density(rnaseq_medians), xlim = c(-1, 15), xlab = "Median gene expression",
     main = "Distribution in current study")
array_medians <- apply(lbb2012_sub, 1, median)
summary(array_medians)
plot(density(array_medians), xlim = c(-1, 15), xlab = "Median gene expression",
     main = "Distribution in Barreiro et al., 2012")
```

Because they have such different distributions, transform each sample to  a standard normal ($~N(\mu=0, \sigma^(2)=1)$).

```{r qqnorm}
counts_cpm_sub_qq <- apply(counts_cpm_sub, 2, function(x) {
  qqnorm(x, plot.it = FALSE)$x})
rownames(counts_cpm_sub_qq) <- rownames(counts_cpm_sub)
rnaseq_medians_qq <- apply(counts_cpm_sub_qq, 1, median)
summary(rnaseq_medians_qq)
plot(density(rnaseq_medians_qq), xlim = c(-4, 4),
     xlab = "Median gene expression",
     main = "Distribution in current study\nafter normalization")
lbb2012_sub_qq <- apply(lbb2012_sub, 2, function(x) {
  qqnorm(x, plot.it = FALSE)$x})
rownames(lbb2012_sub_qq) <- rownames(lbb2012_sub)
array_medians_qq <- apply(lbb2012_sub_qq, 1, median)
summary(array_medians_qq)
plot(density(array_medians_qq), xlim = c(-4, 4),
     xlab = "Median gene expression",
     main = "Distribution in Barreiro et al., 2012\nafter normalization")
```

Combine the two studies.

```{r combine}
exp_combined <- cbind(counts_cpm_sub_qq, lbb2012_sub_qq)
info_comb <- rbind(info[, c("individual", "status", "treatment", "study")],
                   info_lbb)
stopifnot(colnames(exp_combined) == rownames(info_comb))
```

## PCA

```{r calc-pca}
pca <- prcomp(t(exp_combined), scale. = TRUE)
variances <- pca$sdev^2
explained <- variances / sum(variances)
pca_data <- cbind(info_comb, pca$x[, 1:10])
plot(pca, main = "Variance per PC")
```

PC1 versus PC2.

```{r pc1-pc2}
ggplot(pca_data, aes(x = PC1, y = PC2, color = study, size = treatment)) +
  geom_point() +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)),
       title = "PCA before batch correction")
```

PC3 versus PC4.

```{r pc3-pc4}
ggplot(pca_data, aes(x = PC3, y = PC4, color = study, size = treatment)) +
  geom_point() +
  labs(x = sprintf("PC%d (%.2f%%)", 3, round(explained[3] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 4, round(explained[4] * 100, 2)))
```

PC5 versus PC6.

```{r pc5-pc6}
ggplot(pca_data, aes(x = PC5, y = PC6, color = study, size = treatment)) +
  geom_point() +
  labs(x = sprintf("PC%d (%.2f%%)", 5, round(explained[5] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 6, round(explained[6] * 100, 2)))
```

Regress out PC1 to remove effect of study.

```{r regress-pc1}
regressed <- removeBatchEffect(exp_combined, covariates = pca_data$PC1)
```

Perform pca again.

```{r calc-pca-2}
pca2 <- prcomp(t(regressed), scale. = TRUE)
variances2 <- pca2$sdev^2
explained2 <- variances2 / sum(variances2)
pca_data2 <- cbind(info_comb, pca2$x[, 1:10])
plot(pca2, main = "Variance per PC")
```

PC1 vs. PC2.

```{r pc1-pc2-regressed}
ggplot(pca_data2, aes(x = PC1, y = PC2, color = study, size = treatment)) +
  geom_point() +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained2[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained2[2] * 100, 2)),
       title = "PCA after batch correction")
```

Now PC1 is the treatment effect.

## Differential expression

Design matrix Use the group-means parametrization: The mean of each group is
estimated, and the differences are tested with contasts. See Ch. 9 of limma
User's Guide.

```{r limma}
anno <- info_comb
combined <- paste(anno$status, anno$treatment, sep = ".")
combined <- factor(combined, levels = c("contact.none", "tb.none",
                                        "contact.infected", "tb.infected",
                                        "unknown.none", "unknown.infected"))
design <- model.matrix(~0 + combined)
colnames(design) <- gsub("combined", "", colnames(design))
colSums(design)

# Model individual as a random effect
corfit <- duplicateCorrelation(regressed, design, block = anno$individual)
# corfit$consensus
fit <- lmFit(regressed, design, block = anno$individual,
             correlation = corfit$consensus)

# Fit contrasts
contrasts_mat <- makeContrasts(
  # Difference between susceptible and resistant individuals before treatment
  diff_before = tb.none - contact.none,
  # Difference between susceptible and resistant individuals after treatment
  diff_after = tb.infected - contact.infected,
  # Effect of treatment in resistant individuals
  treat_resist = contact.infected - contact.none,
  # Effect of treatment in susceptible individuals
  treat_suscept = tb.infected - tb.none,
  # Difference in effect of treatment between susceptible and resistant
  # individuals
  diff_treat = (tb.infected - tb.none) - (contact.infected - contact.none),
  # Differences between studies before treatment
  study_before = (contact.none) - (unknown.none),
  # Differences between studies after treatment
  study_after = contact.infected - unknown.infected,
  # Effect of treatment in lbb2012
  treat_lbb = unknown.infected - unknown.none,
  levels = design)
fit2 <- contrasts.fit(fit, contrasts_mat)
fit2 <- eBayes(fit2)

# Get model results

get_results <- function(x, number = nrow(x$coefficients), sort.by = "none",
                        ...) {
  # x - object MArrayLM from eBayes output
  # ... - additional arguments passed to topTable
  stopifnot(class(x) == "MArrayLM")
  results <- topTable(x, number = number, sort.by = sort.by, ...)
  return(results)
}

run_ash <- function(x, coef) {
  # Perform multiple testing correction with adaptive shrinkage (ASH)
  #
  # x - object MArrayLM from eBayes output
  # coef - coefficient tested by eBayes
  stopifnot(class(x) == "MArrayLM", coef %in% colnames(x$coefficients))
  result <- ash(betahat = x$coefficients[, coef],
                sebetahat = x$stdev.unscaled[, coef] * sqrt(x$s2.post),
                df = x$df.total)
  return(result)
}

tests <- colnames(fit2$coefficients)
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

for (test in tests) {
  # Extract limma results
  results[[test]] <- get_results(fit2, coef = test)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit2, coef = test)
  results[[test]] <- cbind(results[[test]], lfsr = output_ash$lfsr,
                           lfdr = output_ash$lfdr, qvalue = output_ash$qvalue,
                           svalue = output_ash$svalue)
}
```

## Train a classifier

The training set is the current data.

```{r partition-training}
training <- regressed[, anno$study == "current"]
```

Use leave-one-out-cross-validation.

```{r loocv}
ctrl <- trainControl(method = "LOOCV", classProbs = TRUE, savePred = "final")
```

Find genes which are different between susceptible and resistant individuals at baseline, but not different between studies.

```{r genes-for-classifier}
g <- rownames(training)[results[["diff_before"]]$svalue < .05]
length(g)
genes_for_classifer <- training[g, ]
genes_for_classifer <- t(genes_for_classifer)
genes_for_classifer <- as.data.frame(genes_for_classifer)
genes_for_classifer$status <- anno$status[anno$study == "current"]
genes_for_classifer <- genes_for_classifer[anno$treatment[anno$study == "current"] == "none", ]
```

Fit an elastic net.

```{r glmnet}
fit <- train(status ~ ., data = genes_for_classifer, method = "glmnet",
             trControl = ctrl, metric = "Kappa")
fit
```

Visualize results.

```{r vis}
pred_training <- fit$pred
pred_training <- pred_training[order(pred_training$contact), ]
plot(pred_training$contact, col = ifelse(pred_training$obs == "tb",
                                         "red", "black"),
     ylim = c(0, 1), xlab = "Individual",
     ylab = "Predicted probability of TB resistance",
     main = "Classifier results of current study")
abline(h = 0.9, col = "blue")
```

## Test classifier

Test on lbb2012.

```{r test}
testing <- regressed[g, anno$study == "lbb2012"]
testing <- t(testing)
testing <- testing[anno$treatment[anno$study == "lbb2012"] == "none", ]
fit_lbb <- predict(fit, testing, type = "prob")
fit_lbb
```

Visualize results.

```{r vis-lbb}
pred_testing <- fit_lbb
pred_testing <- pred_testing[order(pred_testing$contact), ]
plot(pred_testing$contact, ylim = c(0, 1), xlab = "Individual",
     ylab = "Predicted probability of TB resistance",
     main = "Classifier results of Barreiro et al., 2012")
abline(h = 0.9, col = "blue")
```

## Compare expression levels between the two studies

```{r compare-exp-level}
training_medians <- apply(genes_for_classifer[-ncol(genes_for_classifer)],
                          2, median)
testing_medians <- apply(testing, 2, median)
plot(training_medians, testing_medians,
     xlab = "Current study", ylab = "Barreiro et al., 2012",
     main = "Comparing median expression levels\nof genes used in classifer")
abline(0, 1, col = "red")
```

## Hierarchical clustering

```{r hclust}
clust_current <- genes_for_classifer[, -ncol(genes_for_classifer)]
labels <- ifelse(grepl("tb", genes_for_classifer$status),
                 "susceptible", "resistant")
labels[labels == "susceptible"] <- paste(labels[labels == "susceptible"],
                                         1:sum(labels == "susceptible"),
                                         sep = ".")
labels[labels == "resistant"] <- paste(labels[labels == "resistant"],
                                         1:sum(labels == "resistant"),
                                         sep = ".")
rownames(clust_current) <- labels
d1 <- dist(clust_current)
h1 <- hclust(d1)
ggdendrogram(h1, rotate = TRUE)
clust_lbb <- testing
labels_lbb <- paste0("Luis.", 1:nrow(clust_lbb))
rownames(clust_lbb) <- labels_lbb
clust_all <- rbind(clust_current, clust_lbb)
d2 <- dist(clust_all)
h2 <- hclust(d2)
ggdendrogram(h2, rotate = TRUE)
```

## Session information

```{r info}
sessionInfo()
```
